lychee-lib/src/collector.rs

```diff
@@ -87,6 +98,20 @@ impl Collector {
         self
     }
 
+    /// Set headers to use when resolving input URLs
+    #[must_use]
+    pub fn headers(mut self, headers: HeaderMap) -> Self {
+        self.headers = headers;
+        self
+    }
+
+    /// Set client to use when resolving input URLs
```

Comment:
Isn't it about checking the URLs?

```suggestion
    /// Set client to use for checking input URLs
```